source/plot/svg/Shapes.ooc

```diff
@@ -1,28 +1,31 @@
 use ooc-math
 
 Shapes: abstract class {
-	line: static func (start, end: FloatPoint2D, color: String) -> String {
-		"<line x1='" clone() & start x toString() & "' y1='" clone() & start y toString() & "' x2='" clone() & end x toString() & "' y2='" clone() & end y toString() & "' stroke-width='1' stroke='" clone() & color clone() & "'/>\n" clone()
+	line: static func (start, end: FloatPoint2D, lineWidth: Float, opacity: Float, color: String) -> String {
+		"<line x1='" << start x toString() >> "' y1='" & start y toString() >> "' x2='" & end x toString() >> "' y2='" & end y toString() >> "' stroke-width='" & lineWidth toString() >> "' stroke='" >> color >> "' stroke-opacity='" & opacity toString() >> "'/>\n"
 	}
-	line: static func ~dashed(start, end: FloatPoint2D, color: String, dashFloatPair: FloatPoint2D) -> String {
-		"<line x1='" clone() & start x toString() & "' y1='" clone() & start y toString() & "' x2='" clone() & end x toString() & "' y2='" clone() & end y toString() & "' stroke-width='1' stroke='" clone() & color clone() & "' stroke-dasharray='" clone() & dashFloatPair toString() & "'/>\n" clone()
+	line: static func ~dashed(start, end: FloatPoint2D, lineWidth: Float, opacity:Float, color: String, dashFloatPair: FloatPoint2D) -> String {
+		"<line x1='" << start x toString() >> "' y1='" & start y toString() >> "' x2='" & end x toString() >> "' y2='" & end y toString() >> "' stroke-width='" & lineWidth toString() >> "' stroke='" >> color >> "' stroke-opacity='" & opacity toString() >> "' stroke-dasharray='" & dashFloatPair toString() >> "'/>\n"
 	}
-	circle: static func (x, y, r, opacity: Float, color: String) -> String {
-		"<circle cx='" clone() & x toString() & "' cy='" clone() & y toString() & "' r='" clone() & r toString() & "' fill='" clone() & color clone() & "' fill-opacity='" clone() & opacity toString() & "'/>\n" clone()
+	circle: static func (center: FloatPoint2D, r, opacity: Float, color: String) -> String {
+		This ellipse(center, r, r, opacity, color)
 	}
 	ellipse: static func (center: FloatPoint2D, rx, ry, opacity: Float, color: String) -> String {
 		"<ellipse cx='" clone() & center x toString() & "' cy='" clone() & center y toString() & "' rx='" clone() & rx toString() & "' ry='" clone() & ry toString() & "' fill='" clone() & color clone() & "' fill-opacity='" clone() & opacity toString() & "'/>\n" clone()
```

Comment:
You shouldn't need to use `clone()` anywhere at all in this file now that you have the new operators...